core/sign_in_test.go

```diff
@@ -0,0 +1,116 @@
+	}
+}
+
+func TestGenerateSignInUriShouldContainReservedScopesByDefault(t *testing.T) {
+	expectedSignInUri := "https://example.com/authorize?client_id=clientId&code_challenge=codeChallenge&code_challenge_method=S256&prompt=consent&redirect_uri=https://example.com/callback&resource=resourceA&resource=resourceB&response_type=code&scope=openid offline_access profile&state=state"
+
+	signInUri, err := GenerateSignInUri(&SignInUriGenerationOptions{
+		AuthorizationEndpoint: "https://example.com/authorize",
+		ClientId:              "clientId",
+		RedirectUri:           "https://example.com/callback",
+		CodeChallenge:         "codeChallenge",
+		State:                 "state",
+		Resources:             []string{"resourceA", "resourceB"},
+		Prompt:                "consent",
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if signInUri != expectedSignInUri {
+		t.Errorf("Expected %s, got %s", expectedSignInUri, signInUri)
+	}
+}
+
+func TestGenerateSignInUriShouldContainReservedScopesAndExtraScopes(t *testing.T) {
+	expectedSignInUri := "https://example.com/authorize?client_id=clientId&code_challenge=codeChallenge&code_challenge_method=S256&prompt=consent&redirect_uri=https://example.com/callback&resource=resourceA&resource=resourceB&response_type=code&scope=openid offline_access profile extra_scope&state=state"
+
+	signInUri, err := GenerateSignInUri(&SignInUriGenerationOptions{
+		AuthorizationEndpoint: "https://example.com/authorize",
+		ClientId:              "clientId",
+		RedirectUri:           "https://example.com/callback",
+		CodeChallenge:         "codeChallenge",
+		State:                 "state",
+		Scopes:                []string{"openid", "offline_access", "profile", "extra_scope"},
+		Resources:             []string{"resourceA", "resourceB"},
+		Prompt:                "consent",
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if signInUri != expectedSignInUri {
+		t.Errorf("Expected %s, got %s", expectedSignInUri, signInUri)
+	}
+}
+
+func TestGenerateSignInUriShouldGenerateUriWithConsentAsThePromptValue(t *testing.T) {
+	expectedSignInUri := "https://example.com/authorize?client_id=clientId&code_challenge=codeChallenge&code_challenge_method=S256&prompt=consent&redirect_uri=https://example.com/callback&resource=resourceA&resource=resourceB&response_type=code&scope=openid offline_access profile&state=state"
+
+	signInUri, err := GenerateSignInUri(&SignInUriGenerationOptions{
+		AuthorizationEndpoint: "https://example.com/authorize",
+		ClientId:              "clientId",
+		RedirectUri:           "https://example.com/callback",
+		CodeChallenge:         "codeChallenge",
+		State:                 "state",
+		Scopes:                []string{"openid", "offline_access", "profile"},
+		Resources:             []string{"resourceA", "resourceB"},
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if signInUri != expectedSignInUri {
+		t.Errorf("Expected %s, got %s", expectedSignInUri, signInUri)
+	}
+}
+
```

Comment:
I think we should add a test case to verify when `prompt` is specified as `Login`

Reply:
updated.